src/lib/journey/callbacks/device-profile/device-profile.svelte

```diff
@@ -0,0 +1,45 @@
+<script lang="ts">
+  import type { SelfSubmitFunction, StepMetadata } from '$journey/journey.interfaces';
+  import { FRDevice, type DeviceProfileCallback } from '@forgerock/javascript-sdk';
+  import type { Maybe } from '$lib/interfaces';
+  import type { ProfileConfigOptions } from '@forgerock/javascript-sdk/lib/fr-device/interfaces';
+  import Spinner from '$components/primitives/spinner/spinner.svelte';
+
+  export let spinner = false;
```

Comment:
Does this need to be exported? Also, I think this component should either render a spinner, when it's the only callback, and nothing when it's composed with other callbacks. That being said, this is internal state and shouldn't be an exposed component prop.

Reply:
yeah i just thought it would be customizable from the component level also but yes the device profile naturally has it